### PR TITLE
Added auto-scaling to eeg graph

### PR DESCRIPTION
**Pull request by ppolk-wave**
- created a new file to handle eeg graph preprocessing
- updated graphs/eeg_viewer to handle the new auto-scaling
- updated streamlit_dashboards/eeg_visualization to scale the df before creating the plotly figure

### CODE REVIEW

File: dsp/graph_preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+
+
+"""
+Low filter units: Time constant in seconds
+0.08 sec is roughly equivalent to a 1.9894 Hz cuttoff frequency
+
+highpass filter to remove signal drift e.g. : (low val=  0.50, high val = None)
+"""
+def apply_waev_filter(mw_object=None, 
+                      time_constant=0.08,
```
Comment (luis-wave) [inline dsp/graph_preprocessing.py:356]: Where does time constant come into play?

File: dsp/graph_preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+        # reattach the 'time' column and combine scaled columns
+        try: scaled_df = pd.concat([times_col, timestamps_col, scaled_eeg], axis=1)
+        except: scaled_df = pd.concat([times_col, scaled_eeg], axis=1)
+
+    return scaled_df
+
+
+"""
+Low filter units: Time constant in seconds
+0.08 sec is roughly equivalent to a 1.9894 Hz cuttoff frequency
```
Comment (luis-wave) [inline dsp/graph_preprocessing.py:351]: Where did you get this info?

File: graphs/eeg_viewer.py [hunk 3]
```diff
@@ -17,15 +17,19 @@ def draw_eeg_graph(df, offset_value, ref):
...
-                x=df["time"],
-                y=df[channel] + offset,  # Apply vertical offset
-                mode="lines",
+                x=df['time'],
+                y=df[channel] + offset,
+                mode='lines',
                 name=channel,
-                line=dict(color="#4E4E4E"),
+                line=dict(
+                    color="black",
```
Comment (luis-wave) [inline graphs/eeg_viewer.py:30]: How does this look in dark mode?

File: dsp/graph_preprocessing.py [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+        # print(f"Time constant used: {time_constant}")
+        if default_time_constant:
+            filters.eeg_filter(mw_object, 1.9894, None)
+        else: 
+            raise Exception("Time constant low filter in progress (from sec)")
+    else:
+        filters.eeg_filter(mw_object, lowf, highf) 
+
+
+"""
```
Comment (luis-wave) [inline dsp/graph_preprocessing.py:372]: Use MyWaveAnalytics, `filters.eeg_filter(self.mw_object, 1, None)`

File: dsp/graph_preprocessing.py [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+        num_channels = num_channels,
+        y_bottom_coordinate = 0,
+        y_top_coordinate = num_channels-1,
+    )
+
+    return format_dict
+
+
+def bipolar_transverse(raw) :
+    # bipolar-transverse montage electrodes
```
[the remark this answers is not in the document]
Comment (luis-wave) [reply dsp/graph_preprocessing.py:451]: from the reference module use `bipolar_transverse_montage`

**Review by luis-wave — CHANGES_REQUESTED** [review]

File: dsp/graph_preprocessing.py [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+    
+    else:
+        raise TypeError("Unsupported data type. Acceptable types: np.ndarray, mne.io.BaseRaw")
+
+# Resample the data in an mne raw object
+def eeg_resample(data=None, freq=100): 
+    # data.load_data()
+    return data.resample(freq)
+
+# Function that calculates the power spectrum and associated frequencies for a 1d or 2d array of data
```
[the remark this answers is not in the document]
Comment (ppolk-wave) [reply dsp/graph_preprocessing.py:424]: this was a placeholder for a function i wanted to write. It has been removed.

File: dsp/graph_preprocessing.py [hunk 7]
```diff
@@ -0,0 +1,528 @@
...
+        eeg_dict = get_referenced_data(raw_mwa, scale_df=scale_df)
+        return eeg_dict
+    else: 
+        return raw_mwa
+
+"""
+All EEG data needed
+"""
+
+def get_referenced_data(raw_mwa=None, scale_df=False):
```
[the remark this answers is not in the document]
Comment (ppolk-wave) [reply dsp/graph_preprocessing.py:92]: I removed all but the necessary scale_dataframe() function necessary for auto-scaling. I will provide these changes in future pull requests with the relevant features

File: dsp/graph_preprocessing.py (hunk 2 above)
Comment (ppolk-wave) [reply dsp/graph_preprocessing.py:351]: that was a quick chatgpt ask, it is not being used in this pull request anyways and wont be in any future ones as looking into the problem more showed this to be inaccurate

File: dsp/graph_preprocessing.py (hunk 1 above)
Comment (ppolk-wave) [reply dsp/graph_preprocessing.py:356]: it is not used in this pull request and wont be used when the relevant feature is presented for code review, in the future

File: dsp/graph_preprocessing.py (hunk 4 above)
Comment (ppolk-wave) [reply dsp/graph_preprocessing.py:372]: I removed code that filtered and processed the data (other than the scaling) as it does pertain to this pull request

File: dsp/graph_preprocessing.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ppolk-wave) [reply dsp/graph_preprocessing.py:451]: I will look into this when I add additional montages as part of a future pull request. I believe the MyWave BPT montage order is different than what we use which is why I made my own

File: graphs/eeg_viewer.py (hunk 3 above)
Comment (ppolk-wave) [reply graphs/eeg_viewer.py:30]: I changed to "#4E4E4E" so its visible in dark mode, thank you!

**Comment by ppolk-wave** [comment]
I initially submitted a graph_processing.py file that had functions for features not yet worked on and not in this pull request. They have been removed and any feedback pertaining to this pull request should be implemented now.

**Review by luis-wave — APPROVED** [review]

**Pull request merged**